src/pages/patientView/clinicalInformation/CohortVariantCountCache.ts

```diff
@@ -0,0 +1,197 @@
+    constructor(private mutationGeneticProfileId:string) {
+        this._cache = Immutable.from<VariantCountCacheType>({
+            numberOfSamples: null,
+            mutationInGene: {},
+            keyword: {}
+        });
+    }
+    public get cache() {
+        return this._cache;
+    }
+
+    public async populate(entrezToKeywordList:EntrezToKeywordList) {
+        const missing = this.getMissing(entrezToKeywordList);
+        this.markPending(missing);
+        try {
+            const data:VariantCount[] = await this.fetch(missing, this.mutationGeneticProfileId);
+            this.putData(missing, data);
+            return true;
+        } catch (err) {
+            this.unmarkPending(missing);
+            return false;
+        }
+    }
+
+    private pendingData() {
+        // For testing
+        const ret = [];
+        for (const entrez of Object.keys(this._cache.mutationInGene)) {
+            if (this._cache.mutationInGene[entrez] && this._cache.mutationInGene[entrez]!.status === "pending") {
+                ret.push(entrez);
+            }
+        }
+        for (const kw of Object.keys(this._cache.keyword)) {
+            if (this._cache.keyword[kw] && this._cache.keyword[kw]!.status === "pending") {
+                ret.push(kw);
+            }
+        }
+        return ret;
+    }
+
+    protected fetch(entrezToKeywordList:EntrezToKeywordList,
+                  mutationGeneticProfileId:string):Promise<VariantCount[]> {
+        let variantCountIdentifiers:VariantCountIdentifier[] = [];
+        for (const entrez of Object.keys(entrezToKeywordList)) {
+            const entrezGeneId = parseInt(entrez, 10);
+            const keywordList = entrezToKeywordList[entrezGeneId];
+            if (keywordList.length === 0) {
+                variantCountIdentifiers.push({
+                    entrezGeneId
+                } as VariantCountIdentifier);
+            } else {
+                variantCountIdentifiers = variantCountIdentifiers.concat(
+                    keywordList.map((keyword:string)=>{
+                        return {
+                            entrezGeneId,
+                            keyword
+                        };
+                    })
+                );
+            }
+        }
+        if (variantCountIdentifiers.length > 0) {
+            return client.fetchVariantCountsUsingPOST({
+                geneticProfileId: mutationGeneticProfileId,
+                variantCountIdentifiers
+            });
+        } else {
+            return Promise.resolve([]);
+        }
+    }
+
+    private getMissing(entrezToKeywordList:EntrezToKeywordList):EntrezToKeywordList {
+        const cache = this._cache;
+        const ret:EntrezToKeywordList = {};
+        for (const entrez of Object.keys(entrezToKeywordList)) {
+            const entrezGeneId = parseInt(entrez, 10);
+            const missingKeywords = entrezToKeywordList[entrezGeneId].filter((kw:string)=>!cache.keyword[kw]);
```

Comment:
i think that Array.includes:Bool method would achieve this.  Babel polyfill allows us to use it in older browsers.